recipe_scrapers/quitoque.py

```diff
@@ -0,0 +1,146 @@
+                }
+                waitingTime
+            }
+        }
+    }
+}
+        """
+        data_body = {
+            "operationName": "getRecipe",
+            "variables": {"id": recipe_id},
+            "query": data_query,
+        }
+        self.data = requests.post(
+            data_url,
+            json=data_body,
+            proxies=proxies,
+            timeout=timeout,
+        ).json()["data"]
+
+    @classmethod
+    def host(cls):
+        return "quitoque.fr"
+
+    def title(self):
+        return self.data["recipe"]["name"]
+
+    def category(self):
+        categories = ""
+        for category in self.data["recipe"]["facets"]:
+            categories += f'{category["name"]}, '
+        return categories[:-2]
+
+    def total_time(self):
+        return self.data["recipe"]["pools"][0]["cookingModes"][0]["waitingTime"]
+
+    def yields(self):
+        return f'{self.data["recipe"]["pools"][0]["nbPerson"]} portions'
+
+    def image(self):
+        return self.data["recipe"]["image"]
+
+    def ingredients(self):
+        ingredients = []
+        stacks = self.data["recipe"]["pools"][0]["cookingModes"][0]["stacks"]
+        for ingredient in stacks["ingredients"]:
+            if ingredient["quantity"] > 0:
+                ingredients.append(
+                    f'{ingredient["literalQuantity"]} {ingredient["product"]["name"]}'
+                )
+            else:
+                ingredients.append(ingredient["product"]["name"])
+        for cupboard_ingredients in stacks["cupboardIngredients"]:
+            if cupboard_ingredients["quantity"] > 0:
+                ingredients.append(
+                    f'{cupboard_ingredients["literalQuantity"]} {cupboard_ingredients["product"]["name"]}'
+                )
+            else:
+                ingredients.append(cupboard_ingredients["product"]["name"])
+        return ingredients
+
+    def instructions(self):
+        instruction = ""
+        steps = self.data["recipe"]["pools"][0]["cookingModes"][0]["steps"]
+        for step in steps:
+            instruction += step["description"].replace("\r", "").replace("\xa0", " ")
+        return instruction
+
+    def nutrients(self):
+        nutritional_informations = self.data["recipe"]["nutritionalInformations"][0]
+        nutrients = {
+            "calories": f'{nutritional_informations["kiloCalorie"]} calories',
+            "fatContent": f'{nutritional_informations["fat"]} grammes',
+            "saturatedFatContent": f'{nutritional_informations["saturatedFat"]} grammes',
+            "carbohydrateContent": f'{nutritional_informations["carbohydrate"]} grammes',
+            "sugarContent": f'{nutritional_informations["sugarCarbohydrate"]} grammes',
+            "fiberContent": f'{nutritional_informations["fiber"]} grammes',
+            "proteinContent": f'{nutritional_informations["protein"]} grammes',
+            "sodiumContent": f'{nutritional_informations["salt"]} grammes',
+        }
```

Comment:
This seems repetitive - it's OK-ish, but it's not ideal that it may need to be updated if the list of supported/available nutrients changes.

Is there a way for a graphQL query to express that _all_ fields from a certain node/element (in this case, `nutritionalInformations`) are requested?

Reply:
You seem more familiar with graphQL than I do. So yes I just checked and this is possible in the request to make something like this : 

```graphql
query getRecipe($id: ID!) { 
    recipe(id: $id) { 
        nutritionalInformations {
            calorie: kiloCalorie
       }
}
```

this query responds as simple as : 

```json
{
    "data": {
        "recipe": {
            "nutritionalInformations": [
                {
                    "calorie": 601.84
                }
            ]
        }
    }
}
```

But unfortunatly I cannot append text to values. So we still need to add text like `{nutritional_informations["kiloCalorie"]} calories`. 

Also as these values are int so there will be returned as is.

Please let me know which solution you preferer.

Regards, 
Romain.